lib/Deserializer/functions.php

```diff
@@ -295,3 +295,42 @@ function mixedContent(Reader $reader) : array {
     return $content;
 
 }
+
+/**
+ * The functionCaller deserializer turns an xml element into whatever your callable return.
+ *
+ * You can use, e.g., a named constructor (factory method) to create an object using
+ * this function.
+ *
+ * @return mixed
+ */
+function functionCaller(Reader $reader, callable $func, string $namespace) {
```

Comment:
Maybe $factory instead of $func

Reply:
This was @evert suggestion actually. In fact, this callable can be used as a factory, but this function is more generic than that. We could use this callable to return an array (or anything else), and imo $factory is more related to objects/classes.